src/ctap/timed_permission.rs

```diff
@@ -12,41 +12,53 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use libtock_drivers::timer::{ClockValue, Duration};
+use super::clock::ClockInt;
+use super::clock::CtapInstant;
+use embedded_time::duration::Milliseconds;
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Debug)]
 pub enum TimedPermission {
     Waiting,
-    Granted(ClockValue),
+    Granted(CtapInstant),
+}
+
+impl Clone for TimedPermission {
+    fn clone(&self) -> Self {
+        match self {
+            Self::Waiting => Self::Waiting,
+            Self::Granted(instant) => Self::Granted(*instant),
+        }
+    }
 }
 
+impl Copy for TimedPermission {}
+
 impl TimedPermission {
     pub fn waiting() -> TimedPermission {
         TimedPermission::Waiting
     }
 
-    pub fn granted(now: ClockValue, grant_duration: Duration<isize>) -> TimedPermission {
-        TimedPermission::Granted(now.wrapping_add(grant_duration))
+    pub fn granted(now: CtapInstant, grant_duration: Milliseconds<ClockInt>) -> TimedPermission {
+        now.checked_add(grant_duration)
```

Comment:
This change might need more testing. We will probably end up with a custom ClockInt to solve all issues around this? Assuming that happens, is a `saturating_add` an option, so less calls to `granted` don't actually do anything? Wrapping clocks are not super rare events with 24 bit actually. A graceful way to handle wrapping would be nice. The implementation right now tries to work around that.

Reply:
Discussed offline, this file can stay as is. Testing should happen in clock.rs, and this file gets looked at later.

Reply:
Since I expect this type to go away sooner than later, I'm happy with any choice that doesn't show worse behavior on real hardware. So if what Julien says works, fine with me!

Reply:
I see 2 immediate options before merging here:
- Use `TimedPermission::Granted(now.checked_add(grant_duration).unwrap())`
- Add `TODO: Should panic or saturate if the grant duration is too big.`